packages/iTwinUI-react/src/core/ComboBox/helpers.ts

```diff
@@ -5,56 +5,117 @@
 import React from 'react';
 import { SelectOption } from '../Select/Select';
 
-type ComboBoxAction = 'open' | 'close' | 'select' | 'focus';
+export const ComboBoxRefsContext = React.createContext<
+  | {
+      inputRef: React.RefObject<HTMLInputElement>;
+      menuRef: React.RefObject<HTMLUListElement>;
+      toggleButtonRef: React.RefObject<HTMLSpanElement>;
+      optionsExtraInfoRef: React.MutableRefObject<
+        Record<string, { __originalIndex: number }>
+      >;
+    }
+  | undefined
+>(undefined);
+ComboBoxRefsContext.displayName = 'ComboBoxRefsContext';
 
 export const comboBoxReducer = (
-  state: { isOpen: boolean; selectedIndex: number; focusedIndex: number },
-  [type, value]: [ComboBoxAction] | [ComboBoxAction, number | undefined],
+  state: {
+    isOpen: boolean;
+    selected: number | number[];
+    focusedIndex: number;
+  },
+  action:
+    | { type: 'multiselect'; value: number }
```

Comment:
might want to save this type because it's reused down below in line 127. Could create `ComboBoxAction` similar to before with this new value

Reply:
Added ComboBoxAction